src/main/java/com/mojang/brigadier/CommandDispatcher.java

```diff
@@ -391,16 +393,23 @@ private ParseResults<S> parseNodes(final CommandNode<S> node, final StringReader
                     return new ParseResults<>(context, parse.getReader(), parse.getExceptions());
                 } else {
                     final ParseResults<S> parse = parseNodes(child, reader, context);
+                    if (unique) {
+                        return parse;
+                    }
                     if (potentials == null) {
-                        potentials = new ArrayList<>(1);
+                        potentials = new ArrayList<>(relevantNodes.size());
```

Comment:
Since `potentials` will contain `ParseResults<S>` up to the amount of `relevantNodes.size()`, allocate that capacity in advance to ensure resizing never occurs.